### ISSUE TITLE
SDK 45 (release): Expo Dev Menu is not showing up, the default RN menu pops up instead

### ISSUE THREAD

**Issue by hirbod** [expo/expo#17373]
### Summary

SDK 45 finally fixed all related mmkv, jsi-helper and expo-push-notification crashes when the expo dev menu popped out.
Reloading is also possible, but the custom dev menu is not showing anymore, instead the default stock ActionSheet by RN.

Personally speaking, I can live with that, because that related issues are gone, but since this is not the intended behavior, I thought it would be helpful to issue a ticket.

Related: https://github.com/expo/expo/pull/17336
Issue: https://github.com/expo/expo/issues/17283

### Managed or bare workflow? If you have made manual changes inside of the `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

iOS

### Package versions

0.9.4

### Environment

```
  expo-env-info 1.0.3 environment info:
    System:
      OS: macOS 12.3.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.19.1 - /usr/local/opt/node@14/bin/node
      Yarn: 1.22.17 - /usr/local/bin/yarn
      npm: 6.14.16 - /usr/local/opt/node@14/bin/npm
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Android Studio: 4.1 AI-201.8743.12.41.6953283
      Xcode: 13.3.1/13E500a - /usr/bin/xcodebuild
    npmGlobalPackages:
      eas-cli: 0.51.0
      expo-cli: 5.4.3
    Expo Workflow: bare
```

### Reproducible demo

None

**Comment by lukmccall** [comment]
The problem with crashes (you referred to) will be finally resolved by https://github.com/expo/expo/pull/17377.

In the newest dev-client version, our auto-setup on iOS doesn't work. We're currently working on fixing that. However, I will share with you a workaround. The only thing you have to do is to set the `DEBUG` value in Swift. To do it you have to insert the `-DDEBUG` flag into the `OTHER_SWIFT_FLAGS` (like here https://stackoverflow.com/a/32686438).

**Comment by tsapeta** [comment]
The fix for this issue has been published in `expo@45.0.1` 🙂

**Comment by hirbod** [comment]
I can confirm that it's working. Reloading is also working, no crashes with notifications or third party JSI libs. GREAT. Finally I can spend less time restarting the app and focus developing. Thanks to everybody involved, this does help a lot!

Fun fact: dev-client 0.9.5 with Expo SDK 45 (without dev menu and server picker) was consuming up to 100MB less memory. After your fixes, it does load everything again correctly and it does consume a big chunk of RAM :D
